proto/storage/cloud_source.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+option go_package = "storage";
+option java_package = "io.stackrox.proto.storage";
+
+import "gogoproto/gogo.proto";
+
+package storage;
+
+message CloudSource {
+    string id                            = 1 [(gogoproto.moretags) = 'sql:"pk,type(uuid)"'];
+    string name                          = 2 [(gogoproto.moretags) = 'sql:"unique"'];
+    string type                          = 3;
+    bool   skip_test_integration         = 4;
+
+    oneof Config {
+        PaladinCloudConfig paladin_cloud = 5;
+        OCMConfig          ocm           = 6;
+    }
+}
+
+message PaladinCloudConfig {
+    string endpoint = 1 [(gogoproto.moretags) = 'scrub:"dependent" validate:"nolocalendpoint"'];
```

Comment:
nit: depending on whether we do the validation of the proto either for the `v1` or the `storage` proto, we may want to drop the validate tag for either one, but it's not strictly required (looking at other samples, this is typically handled by the service, so it may still be the `v1` version, and conversion to `v1` to `storage` is then only done if `v1` has been deemed as "valid").